extensions/eclipse-che-theia-testing-service-ext/src/browser/languages-test-api.ts

```diff
@@ -0,0 +1,186 @@
+/********************************************************************************
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ********************************************************************************/
+
+import { PluginHandleRegistry } from './plugin-handle-registry';
+import { interfaces } from 'inversify';
+import { TestAPI } from '../common/test-protocol';
+import {
+    CompletionContext,
+    CompletionResultDto,
+    SignatureHelp,
+    Hover,
+    DocumentHighlight,
+    Range,
+    TextEdit,
+    FormattingOptions,
+    Definition,
+    DefinitionLink,
+    DocumentLink,
+    CodeLensSymbol,
+    DocumentSymbol,
+    ReferenceContext,
+    Location,
+    SignatureHelpContext,
+    CodeActionContext,
+    CodeAction,
+    FoldingRange,
+} from '@theia/plugin-ext/lib/common/plugin-api-rpc-model';
+import { UriComponents } from '@theia/plugin-ext/lib/common/uri-components';
+import { CancellationToken, FoldingContext } from '@theia/plugin';
+import { SymbolInformation } from 'vscode-languageserver-types';
+import {
+    Position,
+    Selection,
+    RawColorInfo,
+    WorkspaceEditDto
+} from '@theia/plugin-ext/lib/common/plugin-api-rpc';
+
+/**
+ * This class redirects language api requests to the correct sidecars and returns the results
+ */
+export class TestAPIImpl implements TestAPI {
+
+    private readonly pluginHandleRegistry: PluginHandleRegistry;
+
+    constructor(container: interfaces.Container) {
```

Comment:
Is there a reason we're not using injection here?

Reply:
It's just because of how the class is created https://github.com/eclipse/che-theia/pull/597/files#diff-0d93537fbf286e960396f1ccef10a48aR21 and following how its been done in other instances of che-theia